templates/15-rust.sh

```diff
@@ -94,15 +92,21 @@ build_rust_build() {
 			|| abdie "Compilation failed: $?."
 	else
 		abinfo 'Using fallback build method ...'
-	  if [[ "${CARGO_AFTER[@]}" =~ '-p' ]]; then
-		workspace_flag=''
-	  fi
-	  cargo build "${DEFAULT_CARGO_CONFIG[@]}" \
+		# determine if buiding a package
+		local workspace_flag
+		workspace_flag="--workspace"
+		for element in "${CARGO_AFTER[@]}"; do
+			if [[ "$element" == "-p" || "$element" == "--package" ]]; then
+				unset workspace_flag
+				break
+			fi
+		done
+		cargo build "${DEFAULT_CARGO_CONFIG[@]}" \
 			--release --locked \
-			"${CARGO_AFTER[@]}" "${workspace_flag}" \
+			"${CARGO_AFTER[@]}" ${workspace_flag} \
```

Comment:
Point taken for this and the indentation fix below, but these fixes should be in their own commits (same for the if branch above, I think you should just ditch the for loop change above, it seems unnecessary).

Reply:
That seems a repeated work (because some code will finally be deleted) but I will try to rebase to two commits if desired.

Reply:
> That seems a repeated work (because some code will finally be deleted) but I will try to rebase to two commits if desired.

Repeated work is still not "work in one commit."